eland/operations.py

```diff
@@ -1509,6 +1571,49 @@ def batch_size() -> None:
     def show_progress(self) -> bool:
         return self._show_progress
 
+class PandasDataFrameIterator(ABC):
+    def __init__(self, query_compiler, es_results_generator, post_processing):
+        self._query_compiler = query_compiler
+        self._es_results_generator = es_results_generator
+        self._post_processing = post_processing
+
+    def __iter__(self):
+        return self
+
+    @abstractmethod
+    def __next__(self):
+        pass
+
+    def _es_results_to_pandas(self, es_results):
+        _, df = self._query_compiler._es_results_to_pandas(es_results)
+        df = Operations._apply_df_post_processing(df, self._post_processing)
+        return df
+
+class PandasDataFrameRowsIterator(PandasDataFrameIterator):
+    def __init__(self, query_compiler, es_results_generator, post_processing):
+        super().__init__(query_compiler, es_results_generator, post_processing)
+
+    def __next__(self):
+        es_result = next(self._es_results_generator) # This will return one es result in each iteration
+        if es_result:
+            df = self._es_results_to_pandas([es_result]) # So, it need to be converted to list
```

Comment:
So, `_es_results_to_pandas` is the important method. 
Its logic is to initialize a `FieldMappingCache` on the first document. and reuse it for all the subsequent documents.

If we pass a list with one document every time, it becomes very slow. The whole reason why cache exists goes away. 

My suggestion is to Convert  `_es_results_to_pandas` into a generator itself. which yields each document, so the state is preserved. 

So, it looks like
generator (either tuples or rows) => generator(_es_results_to_pandas) -> generator (_es_results_generator) -> ES


So, in this way only required documents are generator per yield. Even though if someone wants to break the chain, it will be efficient by not querying entire index.

Reply:
Great suggestion, totally agree with this! :muscle:

Reply:
This is a good idea!

I try to convert `QueryCompiler._es_results_to_pandas()` into a generator itself. And did some speed tests.

I used a data set of 50,000 rows to do the test, and the results are as follows:

Before conversion:
```
ed.iterrows(),  It took a total of `2 minutes 30 seconds` after the iteration
ed.itertuples(), It took a total of `3 minutes 53 seconds` after the iteration
ed.to_pandas(), It took `15 seconds`
```
After conversion:
```
ed.iterrows(),  It took a total of `1 minutes 52 seconds` after the iteration
ed.itertuples(), It took a total of `2 minutes 11 seconds` after the iteration
ed.to_pandas(), It took `1 minutes 54 seconds`
```

Reply:
`iterrows()` and `itertuples()` are both faster.

But `to_pandas()` has slowed down, and there should be other room for optimization. 

I would like to ask if there are more suggestions, Thanks very much.